File: 2015/01_not_quite_lisp/part1.py
with open("input.txt", "r") as file:
    i = 0
    for line in file:
        for char in line:
            if char == "(":
                i += 1
            elif char == ")":
                i -= 1
            else:
                print("DEAD")
                exit(1)
    print(i)

File: 2015/01_not_quite_lisp/part1.go
package main

import (
	"bufio"
	"fmt"
	"os"
)

func main() {
	input, err := os.Open("input.txt")
	if err != nil {
		fmt.Println(err)
	}
	defer input.Close()

	scanner := bufio.NewScanner(input)
	for scanner.Scan() {
		line := scanner.Text()
		i := 0
		for _, c := range line {
			if c == 40 {
				i += 1
			} else if c == 41 {
				i -= 1
			}
		}
		fmt.Println(i)
	}

	if err := scanner.Err(); err != nil {
		fmt.Println(err)
	}
}
